service/src/main/kotlin/fi/espoo/evaka/reports/HolidayPeriodAttendanceReport.kt

```diff
@@ -322,29 +333,40 @@ private fun Database.Read.getReservationsForChildrenOverRange(
         )
         .groupBy { Pair(it.date, it.childId) }
 
-private fun Database.Read.getServiceNeedOccupancyInfoOverRangeForUnit(
-    daycareId: DaycareId,
-    range: FiniteDateRange,
-) =
-    getServiceNeedOccupancyInfoOverRange(
-        Predicate { where("pl.unit_id = ${bind(daycareId)}") },
-        range,
-    )
-
 private fun Database.Read.getServiceNeedOccupancyInfoOverRangeForChildren(
     childIds: Set<PersonId>,
     range: FiniteDateRange,
 ) =
     if (childIds.isEmpty()) emptyList()
     else
         getServiceNeedOccupancyInfoOverRange(
-            Predicate { where("pl.child_id = ANY (${bind(childIds)})") },
             range,
+            placementPred = Predicate { where("pl.child_id = ANY (${bind(childIds)})") },
+            groupsPred = Predicate.alwaysTrue(),
+        )
+
+private fun Database.Read.getServiceNeedOccupancyInfoOverRangeForGroups(
+    groupIds: Set<GroupId>,
+    unitId: DaycareId,
+    range: FiniteDateRange,
+): List<ChildServiceNeedOccupancyInfo> =
+    if (groupIds.isEmpty())
+        getServiceNeedOccupancyInfoOverRange(
+            range,
+            placementPred = Predicate { where("pl.unit_id = ${bind(unitId)}") },
```

Comment:
Täällä pitäisi vielä käyttää `$it` (eli tässä `$it.unit_id`) jotta taulun nimen parametrisointi toimii, ja sama idea muissakin predikaateissa ylempänä ja alempana

Reply:
Nyt myös nämä muutettu koko raporttiin.